src/libraries/tests.proj

```diff
@@ -43,6 +43,12 @@
   
   <Target Name="Build" DependsOnTargets="BuildAllProjects" />
   <Target Name="BuildAllProjects">
+    <!-- These projects are required to restore the runtime and setup testhost -->
+    <MSBuild Targets="Build"
+             Projects="$(MSBuildThisFileDirectory)restore\runtime\runtime.depproj;$(MSBuildThisFileDirectory)\pretest.proj"
```

Comment:
I think this should only be done on CI. Locally we run `runtime.depproj` as part of the restore phase. 

I would add an `ItemGroup` to do this, something like:
```xml
<ItemGroup>
  <SetupProjects Include="$(MSBuildThisFileDirectory)\pretest.proj" />
  <SetupProjects Include="$(MSBuildThisFileDirectory)restore\runtime\runtime.depproj" />
</ItemGroup>

<MSBuild Targets="Build"
                 Projects="@(SetupProjects)
                Condition="'$(ContinuousIntegrationBuild)' == 'true'" 
                ... />
```

cc: @ViktorHofer

Reply:
done